examples/generic/generic_systems.rs

```diff
@@ -0,0 +1,11 @@
+use bevy::ecs::component::Component;
+
+// remove all entites that aren't the player
+fn remove_entities<T: Component>(q: Query<Entity, With<T>>) {
+    for entity in q.iter() {
+        let id = Entity;
+        if entity != id.0 {
+            command.entity(entity).despawn();
+        }
+    }
+}
```

Comment:
So far this seems like a good starting point!
However, it'll definitely need some fleshing out. 

A few logistical things,
1) This will need to be added as an `[[example]]` in the primary `Cargo.toml` file
2) This will need to be added in the `examples/README.md` 

Now with a few more suggestions
1) All example are runnable, meaning they have a `fn main()` associated with them so that a user is able to run the example.
    - So having a `fn main()` along with an `App::build().add_system(MyGenericSystem<T>)` will be needed.
2) Since this is covering `generic` system, we need to highlight the primary reason we'd use them. And that is _re-usability_.
    - So ideally, you would have 1 generic system, and then you would add it it at least 2 times to the primary application with a different generic type for each addition. 

for example
```rust
struct A;
struct B;

fn system<T>(_: Query<&T>) {}

fn main() {
    App::build()
        .add_system(system<A>)
        .add_system(system<B>)
        .run();
}
```

3) While the `remove_entity` system you provided was a great start, there won't actually be an entities alive to remove making the system do nothing. 
    - It would be nice to have the system do something the the user can view. Perhaps this is a `spawn_entity<T>`, or a `log_component<T>` for example.

Again great start and please ask if you have any questions along the way! :smile: